### DISCUSSION TITLE
Can we delay calculation of FileSystemPaths

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#7314] [Q&A]
In https://github.com/Jermolene/TiddlyWiki5/issues/7151#issuecomment-1364684348 I learned that

> it is `core/modules/utils/filesystem.js`'s `generateTiddlerFilepath` using filters from FileSystemPath.tid

This can be slow. And the comment is

> Generate the filepath for saving a tiddler

And in `core/modules/commands/savewikifolder.js`'s `saveTiddler` it uses `$:/config/FileSystemPaths` again, to "Generate the filepath for saving a tiddler"

So I think the one in `core/modules/utils/filesystem.js` might be useless, and makes startup time dramatically increased. This may be a point we can optimize, but I'm not sure.

**Comment by Jermolene** [comment]
Hi @linonetwo the code in `core/modules/commands/savewikifolder.js` is the implementation of the `--savewikifolder` command, and is not invoked unless that command is used.

In the normal course of events, `generateTiddlerFilepath` is only invoked once for each tiddler that is synced back to the filesystem.

**Reply by pmario** [reply]
Is it possible that `$:/config/OriginalTiddlerPaths` causes a delay on the server for @linonetwo because of the huge number of tiddlers?
